packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -819,6 +819,88 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     }
   }
 
+  async buildGraphForQuery(
+    graph: PureModel,
+    entities: Entity[],
+    buildState: ActionState,
+    options?: GraphBuilderOptions,
+    _report?: GraphManagerOperationReport,
+  ): Promise<void> {
+    const stopWatch = new StopWatch();
+    const report = _report ?? createGraphBuilderReport();
+    buildState.reset();
+
+    try {
+      // deserialize
+      buildState.setMessage(`Deserializing elements...`);
+      const data = new V1_PureModelContextData();
+      await V1_entitiesToPureModelContextData(
+        entities,
+        data,
+        this.pluginManager.getPureProtocolProcessorPlugins(),
+        this.subtypeInfo,
+      );
+      stopWatch.record(
+        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_DESERIALIZE_ELEMENTS__SUCCESS,
+      );
+
+      // prepare build inputs
+      const buildInputs: V1_PureGraphBuilderInput[] = [
+        {
+          model: graph,
+          data: V1_indexPureModelContextData(
+            report,
+            data,
+            this.graphBuilderExtensions,
+          ),
+        },
+      ];
+
+      // index
+      buildState.setMessage(
+        `Indexing ${report.elementCount.total} elements...`,
+      );
+      await this.initializeAndIndexElements(graph, buildInputs, options);
+      stopWatch.record(
+        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_INDEX_ELEMENTS__SUCCESS,
+      );
+
+      // build types
+      buildState.setMessage(`Building domain models...`);
+      await this.buildTypes(graph, buildInputs, options);
+      stopWatch.record(
```

Comment:
necessary elements: 
class, function, service, 
(path of runtime, connection, store)

Reply:
await this.buildOtherElements(graph, buildInputs, options);
      stopWatch.record(
        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_BUILD_OTHER_ELEMENTS__SUCCESS,
      );

to resolve dataspace executables